drivers/sensor/ti_hdc/Kconfig

```diff
@@ -8,6 +8,6 @@
 
 menuconfig TI_HDC
 	bool "Texas Instruments Temperature and Humidity Sensor (e.g. HDC1008)"
-	depends on I2C && HAS_DTS_I2C && GPIO && HAS_DTS_GPIO
+	depends on I2C && HAS_DTS_I2C
```

Comment:
Why are you removing this dependency?

Reply:
Because it depends on the variation of the ti_hdc sensor and the GPIO interrupt functionality is optional. Of course, if you have a better idea I can change it.